Mage.Tests/src/test/java/org/mage/test/cards/abilities/oneshot/KaghaShadowArchdruidAbilityTest.java

```diff
@@ -0,0 +1,167 @@
+package org.mage.test.cards.abilities.oneshot;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+import mage.abilities.Ability;
+import mage.abilities.keyword.DeathtouchAbility;
+import mage.cards.Card;
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+
+public class KaghaShadowArchdruidAbilityTest extends CardTestPlayerBase {
+    
+    // 1. Deathtouch and mill attack trigger
+    @Test
+    public void testKaghaAttackTrigger() {
+        // initialize library, hand and battlefield
+        removeAllCardsFromLibrary(playerA);
+        addCard(Zone.HAND, playerA, "Swamp",7);
+        addCard(Zone.LIBRARY, playerA, "Forest", 4);
+        addCard(Zone.BATTLEFIELD, playerA, "Kagha, Shadow Archdruid", 1);
+
+        setStopAt(1, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        attack(3, playerA, "Kagha, Shadow Archdruid");
+        setStopAt(3, PhaseStep.END_TURN);
```

Comment:
Include `setStrictChooseMode(true)` for all new tests. (This avoids potential discrepancies being silently chosen by the AI and makes the tests more robust.)

Reply:
Thanks, done!